Sources/Client/Model/Channel/ChannelResponse.swift

```diff
@@ -76,23 +76,29 @@ public struct ChannelResponse: Decodable {
         
         let unreadMessageRead = userUnreadMessageRead()
         channel.unreadMessageReadAtomic.set(unreadMessageRead)
+        
         var unreadCount = ChannelUnreadCount.noUnread
         let currentUser = Client.shared.user
         
         if let unreadMessageRead = unreadMessageRead {
-            for message in messages.reversed() {
-                if message.user.isCurrent {
-                    continue
-                }
-                
-                if message.created > unreadMessageRead.lastReadDate {
-                    unreadCount.messages += 1
+            // Backend sends message unread count, use it directly
+            unreadCount.messages = unreadMessageRead.unreadMessageCount
+            
+            if unreadMessageRead.unreadMessageCount > 0 {
+                // Calculate mentioned message unread count
+                // This is approximate since it'll be limited for the messages we've fetched
+                for message in messages.reversed() {
+                    if message.user.isCurrent {
+                        continue
+                    }
                     
-                    if message.user != currentUser, message.mentionedUsers.contains(currentUser) {
-                        unreadCount.mentionedMessages += 1
+                    if message.created > unreadMessageRead.lastReadDate {
+                        if message.user != currentUser, message.mentionedUsers.contains(currentUser) {
+                            unreadCount.mentionedMessages += 1
+                        }
+                    } else {
+                        break
                     }
-                } else {
-                    break
                 }
             }
```

Comment:
suggestion:
```swift
                  // Calculate mentioned message unread count
                  // This is approximate since it'll be limited for the messages we've fetched
                  for message in messages.reversed() where !message.user.isCurrent {
                     if message.created > unreadMessageRead.lastReadDate {
                         if message.mentionedUsers.contains(currentUser) {
                             unreadCount.mentionedMessages += 1
                         }
                     } else {
                         break
                     }
                  }
```